File: go-solutions/2390_remove_stars_test.go
package leetcode

import "testing"

type removeStarsTest struct {
	s    string
	want string
}

var removeStarsTests = []removeStarsTest{
	{"leet**cod*e", "lecoe"},
	{"erase*****", ""},
}

func TestRemoveStars(t *testing.T) {
	for _, test := range removeStarsTests {
		if got := RemoveStars(test.s); got != test.want {
			t.Errorf("RemoveStars(%q) = %q, want %q", test.s, got, test.want)
		}
	}
}

func BenchmarkRemoveStars(b *testing.B) {
	for i := 0; i < b.N; i++ {
		RemoveStars("leet**cod*e")
	}
}

File: ts-solutions/tests/2390-remove-stars.test.ts
import { removeStars } from '../src/2390-remove-stars'

type RemoveStringsTest = {
	s: string
	want: string
}

const removeStringsTest: RemoveStringsTest[] = [
	{ s: `leet**cod*e`, want: `lecoe` },
	{ s: `erase*****`, want: `` },
]

describe(`removeStrings`, () => {
	it.each(removeStringsTest)(`removeStrings($s) => $want`, ({ s, want }) => {
		const got = removeStars(s)
		expect(got).toBe(want)
	})
})
